Add an entity fact retriever

iain-b opened the pull request:
Signed-off-by: Iain Billett <iain@roadie.io>

I've added facts which describe if an entity has an owner, description and tags. Also I've allowed the retriever to be configured to create facts for arbitrary annotations. This would happen in the backend application so gives flexibility and avoids any awkward coupling between tech insights and other plugins.

freben committed:
Implement the locks part of the task manager

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
Get most of the task worker code into place

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

Rugvip committed:
core-app-api: deprecate dependency on core-components

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

freben committed:
finishing touches

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
docs

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
do not modify the backend yet

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

Rugvip committed:
core-app-api: deprecate not passing in icons + add defaultAppIcons()

Co-authored-by: Johan Haals <johan.haals@gmail.com>
Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
core-{app,plugin}-api: refactor AppTheme to use provider + defaultAppThemes

Co-authored-by: Johan Haals <johan.haals@gmail.com>
Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

freben committed:
move to a separate package instead and address comments

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
log on failed release

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

Rugvip committed:
core-components: provide withDefaults and work around type duplication

Co-authored-by: Johan Haals <johan.haals@gmail.com>
Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
changesets: updated changesets to use withDefaults

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
core-app-api, create-app, dev-utils: updates to use withDefaults

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
core-*: update API reports

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
core-app-api: remove duplicate AppThemeProvider

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
core-components: tweak exports and update api reports

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

freben committed:
remove knexfile

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
remove the lock functionality

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
get rid of the ts-ignore

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
use dashes in not-ready-yet

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
log the task id as well when unable to parse settings

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
make timeout and frequency mandatory

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

Rugvip committed:
packages: new app-defaults package + move over withDefaults

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
app-defaults: refactor things into folders and move things closer to home

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

freben committed:
use a proper AbortController instead of the home baked one

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
make the original task definition into just one object

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

Rugvip committed:
app-defaults: move AppThemeProvider back to core-app-api

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
app-defaults: move defaultConfigLoader back to core-app-api

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
core-app-api: rename PrivateAppImpl to AppManager

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

freben committed:
make it possible to abort tasks

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
final cleanup, removing unschedule

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
fix api export and test timeouts

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
more tests and a review comment

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

Rugvip committed:
app-defaults,core-app-api: refactor into createApp/createSpecializedApp split

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

freben committed:
rename to TaskScheduler to reduce collision risk with other concepts

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

Rugvip committed:
dev-utils,test-utils,app: update to use app-defaults

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

freben committed:
fix missing link in api report

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

cmpadden committed:
Refactor create-app tasks and introduce regression tests

- Tasks were moved from the script entrypoint to the lib directory
- Doc comments were added for each task function defined in `src/lib/tasks.ts`
- The `yarn test` script was added to package.json
- Unit tests were written for each task -- verying file operations using fs-mock

Signed-off-by: Colton Padden <colton.padden@fastmail.com>

freben committed:
bump to the latest versions of backstage packages

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
remove double config dep in cli

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
update to the right version of the errors package

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

Rugvip committed:
changesets,create-app: update for createApp split

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

freben committed:
remove unnecessary dep on backend-test-utils

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

cmpadden committed:
Add regression tests to createApp entrypoint

Ensure that appropriate tasks are called depending on inquirer prompt answers,
and command-line argument options (--path, and --skip-install).

Signed-off-by: Colton Padden <colton.padden@fastmail.com>

Rugvip committed:
core-components: wrap page stories in test app

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
update API reports for createApp split

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

cmpadden committed:
Add type annotations for mocked child_process.exec

Signed-off-by: Colton Padden <colton.padden@fastmail.com>

cmpadden committed:
Use mock-fs during createApp to prevent filesystem modifications

Signed-off-by: Colton Padden <colton.padden@fastmail.com>

Fox32 committed:
Lazy load `LazyLog` as it is rarely used

Signed-off-by: Oliver Sand <oliver.sand@sda-se.com>

Fox32 committed:
Adjust to review comments

Signed-off-by: Oliver Sand <oliver.sand@sda-se.com>

Fox32 committed:
Use `<Progress>` instead of `<LinearProgress />` in combination with `Suspense`

Signed-off-by: Oliver Sand <oliver.sand@sda-se.com>

iain-b committed:
Add an entity fact retriever

Signed-off-by: Iain Billett <iain@roadie.io>

iain-b commented on plugins/tech-insights-backend/src/service/fact/factRetrievers/entityFactRetriever.test.ts:
```diff
@@ -0,0 +1,111 @@
+    getEntitiesMock.mockClear();
+  });
+
+  describe('hasSpecWithOwner', () => {
```
I'll add more cases pending review of the approach.

iain-b commented on plugins/tech-insights-backend/src/service/fact/factRetrievers/entityFactRetriever.ts:
```diff
@@ -0,0 +1,92 @@
+    const catalogClient = new CatalogClient({
+      discoveryApi: discovery,
+    });
+    const entities = await catalogClient.getEntities({
```
I wonder if it makes sense / was considered to move this logic to the engine and just pass the matching entities to the handler? given the entity filter is a field on the object anyway (I haven't considered this deeply just an idea)

Xantier replied:
It might be better suited in there tbh. The case for not moving it there would be to enable these fact retrievers to get data for items that are not directly tied to the entity structure of Backstage which was one of the questions in the original PRFC. Though moving it there wouldn't prevent it tbh. 

Maybe the filter could be passed in as a context item to the retriever so the implementer can decide whether to use it or not?

iain-b replied:
> Maybe the filter could be passed in as a context item to the retriever so the implementer can decide whether to use it or not?

That's better for sure. Then at least we know there's probably a link between that field and the actual entities polled and is just a bit neater .

iain-b commented on plugins/tech-insights-backend/src/service/fact/factRetrievers/entityFactRetriever.ts:
```diff
@@ -0,0 +1,92 @@
+          name: it.metadata.name,
+        },
+        facts: {
+          hasSpecWithOwner: Boolean(it.spec?.owner),
```
I'm really not sure if this is useful at all given `owner` is a required value in the spec of all the (matching) entities bar template.

iain-b commented on plugins/tech-insights-backend/src/service/fact/factRetrievers/entityFactRetriever.ts:
```diff
@@ -0,0 +1,92 @@
+        facts: {
+          hasSpecWithOwner: Boolean(it.spec?.owner),
+          hasSpecWithGroupOwner: Boolean(it.spec?.owner && it.spec?.owner),
+          hasOwnedByRelation: Boolean(
```
I'm not sure if it makes sense / is necessary to check the relations ? just an implementation detail?

iain-b commented on plugins/tech-insights-backend/src/service/fact/factRetrievers/entityFactRetriever.test.ts:
```diff
@@ -0,0 +1,111 @@
+      kind: 'Component',
+      spec: {
+        type: 'service',
+        lifecycle: 'test',
```
`owner: '', `
^^ this is probably more realistic given it's a required field.

iain-b committed:
Add some more test cases

Signed-off-by: Iain Billett <iain@roadie.io>

Xantier commented:
It's a bit of a thinker this one. Should we store some of these values as raw data instead of doing the check itself on the fact retriever already. Something like amount of relations could be something that may or may not be useful to display somewhere sometime in the future (say, historic data of how an entity usage has grown or sumn) and the check itself for scorecard purposes can be written as `amountOfRelations > 0` or something. 

In any case the implementation of this fact retriever will be super trivial.

Xantier commented:
> seems suuuuper trivial I'm open to other checks on entities

I think when opening this to upstream we should be clear about the terminology here. It's a bit awkward since the implementation is so trivial, but these are `facts` not `checks`

iain-b commented:
> > seems suuuuper trivial I'm open to other checks on entities
> 
> I think when opening this to upstream we should be clear about the terminology here. It's a bit awkward since the implementation is so trivial, but these are `facts` not `checks`

Yep absolutely, IIRC there's no reference to "check" in the code (?) I wrote this comment a bit quickly

iain-b commented:
> It's a bit of a thinker this one. Should we store some of these values as raw data instead of doing the check itself on the fact retriever already. Something like amount of relations could be something that may or may not be useful to display somewhere sometime in the future (say, historic data of how an entity usage has grown or sumn) and the check itself for scorecard purposes can be written as `amountOfRelations > 0` or something.
> 
> In any case the implementation of this fact retriever will be super trivial.

I guess I'm running on the assumption there's some action users might want to take based on fact data. We don't necessarily need to know what that is tbf. When it comes to relations it's hard to say. I can imagine API entities having facts on which (or simply how many) facts they have might be useful in driving completeness given AFAIK users specify that manually ?

Xantier commented:
Here's the original issue btw that we could try to model this from: https://github.com/backstage/backstage/issues/2662

I don't know if any of those items are actually applicable or if we are able to create facts out of them but this could maybe be a partial solution to that highly voted issue (the other part being Irma's scorecard UI itself)

iain-b commented:
> Here's the original issue btw that we could try to model this from: [backstage#2662](https://github.com/backstage/backstage/issues/2662)
> 
> I don't know if any of those items are actually applicable or if we are able to create facts out of them but this could maybe be a partial solution to that highly voted issue (the other part being Irma's scorecard UI itself)

Thanks that's really helpful; Didn't know about this issue. Definitely I could add a description fact, a tags fact... I hesitate around CI/CD plugin annotations as the values of those are defined in the plugins and I don't want to create a dependency to those (I could hard code them; I'd rather not hard code them). Similarly does techdocs still require an annotation ? (IIRC there's now a default?)

Xantier commented:
> Similarly does techdocs still require an annotation ? (IIRC there's now a default?)

I think so. AFAIK that's how they are filtered to the list on the `Docs` table in the frontend at least. (`plugins/techdocs/src/home/components/TechDocsPicker.tsx`)

The CI thing is interesting. I agree we shouldn't hard code those but I'm not certain what is a better approach. Maybe it just needs to be a fact retriever for that specific plugin? I feel like that would be the correct way to do it tbh.

GoFightNguyen committed:
[docs] clarify example showing that parameters types propagate

Signed-off-by: Jason Nguyen <jason-nguyen@pluralsight.com>

Commit:
feat(catalog): remove 'View Api' link from AboutCard

Signed-off-by: Dominik Schwank <dominik.schwank@sda.se>

Commit:
Add prop to override color selection in gauge components

Signed-off-by: James Turley <jamesturley@gocardless.com>

Commit:
Modify exported getProgressColor to accept arguments object

Signed-off-by: James Turley <jamesturley@gocardless.com>

Commit:
Additional normalisation

Signed-off-by: Tim Jacomb <timjacomb1+github@gmail.com>

freben committed:
use lodash.once instead of memoize, and skip unnecessary PK index

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

iain-b committed:
Add some more test cases

Signed-off-by: Iain Billett <iain@roadie.io>

dependabot[bot] committed:
build(deps): bump passport-oauth2 from 1.6.0 to 1.6.1

Bumps [passport-oauth2](https://github.com/jaredhanson/passport-oauth2) from 1.6.0 to 1.6.1.
- [Release notes](https://github.com/jaredhanson/passport-oauth2/releases)
- [Changelog](https://github.com/jaredhanson/passport-oauth2/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jaredhanson/passport-oauth2/compare/v1.6.0...v1.6.1)

---
updated-dependencies:
- dependency-name: passport-oauth2
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @spotify/eslint-config-typescript

Bumps [@spotify/eslint-config-typescript](https://github.com/spotify/web-scripts) from 10.0.0 to 12.0.0.
- [Release notes](https://github.com/spotify/web-scripts/releases)
- [Changelog](https://github.com/spotify/web-scripts/blob/master/CHANGELOG.md)
- [Commits](https://github.com/spotify/web-scripts/compare/v10.0.0...v12.0.0)

---
updated-dependencies:
- dependency-name: "@spotify/eslint-config-typescript"
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Rugvip committed:
Merge pull request #7746 from backstage/mob/snyk-issue-creation-workflow

Opening Github issues for Snyk vulnerabilities

benjdlambert committed:
Merge pull request #7875 from marleypowell/marley/7641-build-table-story

docs: Created stories for BuildTable component to document different component states.

freben committed:
Merge pull request #7895 from backstage/dependabot/npm_and_yarn/passport-oauth2-1.6.1

build(deps): bump passport-oauth2 from 1.6.0 to 1.6.1

Commit:
updated timestamp conversion to use Luxon and adrs-adr012 guideline, added luxon as dependency

Signed-off-by: Jeremy Guarini <jguarini@paloaltonetworks.com>

Commit:
Add Palo Alto Networks to Adopters page

Signed-off-by: Jeremy Guarini <jguarini@paloaltonetworks.com>

freben committed:
Merge pull request #7843 from backstage/joonp/nested-entity-search-filter

Allow singleton and flexibly nested EntityFilters

Commit:
change luxon version to match others

Signed-off-by: Jeremy Guarini <jguarini@paloaltonetworks.com>

dependabot[bot] committed:
build(deps): bump graphiql from 1.4.2 to 1.4.7

Bumps [graphiql](https://github.com/graphql/graphiql) from 1.4.2 to 1.4.7.
- [Release notes](https://github.com/graphql/graphiql/releases)
- [Changelog](https://github.com/graphql/graphiql/blob/main/CHANGELOG.md)
- [Commits](https://github.com/graphql/graphiql/compare/graphiql@1.4.2...graphiql@1.4.7)

---
updated-dependencies:
- dependency-name: graphiql
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

timbonicus committed:
Merge pull request #7904 from jeremyguarini/adopters_update

Add Palo Alto Networks to Adopters page

Commit:
address vale issue with href

Signed-off-by: Jeremy Guarini <jguarini@paloaltonetworks.com>

Rugvip committed:
create-app: default to making new plugins private

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

freben committed:
Merge pull request #7889 from jeremyguarini/code_coverage_ui_updates

Minor UI updates to make numbers and dates a bit more human friendly

Rugvip committed:
Merge pull request #7873 from dehamzah/add-overrides-ui-name

Add Theme Overrides for Sidebar

Rugvip committed:
Merge pull request #7905 from backstage/dependabot/npm_and_yarn/graphiql-1.4.7

build(deps): bump graphiql from 1.4.2 to 1.4.7

dependabot[bot] committed:
build(deps): bump apollo-server from 2.21.0 to 2.25.3

Bumps [apollo-server](https://github.com/apollographql/apollo-server/tree/HEAD/packages/apollo-server) from 2.21.0 to 2.25.3.
- [Release notes](https://github.com/apollographql/apollo-server/releases)
- [Changelog](https://github.com/apollographql/apollo-server/blob/main/CHANGELOG.md)
- [Commits](https://github.com/apollographql/apollo-server/commits/apollo-server@2.25.3/packages/apollo-server)

---
updated-dependencies:
- dependency-name: apollo-server
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
display X-axis in ascending order

Signed-off-by: Jeremy Guarini <jguarini@paloaltonetworks.com>

dependabot[bot] committed:
build(deps): bump @spotify/eslint-config-react from 10.0.0 to 12.0.0

Bumps [@spotify/eslint-config-react](https://github.com/spotify/web-scripts) from 10.0.0 to 12.0.0.
- [Release notes](https://github.com/spotify/web-scripts/releases)
- [Changelog](https://github.com/spotify/web-scripts/blob/master/CHANGELOG.md)
- [Commits](https://github.com/spotify/web-scripts/compare/v10.0.0...v12.0.0)

---
updated-dependencies:
- dependency-name: "@spotify/eslint-config-react"
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

RodionGork committed:
fix: kuber plugin pass id token on getClusters req

Signed-off-by: rgorkovenko <rodiongork@github.com>

marleypowell committed:
Merge branch 'master' of https://github.com/backstage/backstage into marley/7641-consume-exported-ado-types

marleypowell committed:
feat: Added new types to `@backstage/plugin-azure-devops-common`.

Signed-off-by: Marley Powell <Marley.Powell@exclaimer.com>

jhaals committed:
Merge pull request #7861 from jluk-box/core-plugin-api-docs

Improve API documentation for @backstage/core-plugin-api

jhaals committed:
core-plugin-api: Deprecate unused types

Signed-off-by: Johan Haals <johan.haals@gmail.com>

jhaals committed:
api report

Signed-off-by: Johan Haals <johan.haals@gmail.com>

freben committed:
fake the test config to get rid of the timeouts

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

Commit:
Remove mailEnabled filter

Signed-off-by: Tim Jacomb <tim.jacomb@hmcts.net>

freben committed:
get rid of config loading here too

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
Merge pull request #7914 from backstage/freben/proxy-router-test

fake the test config to get rid of the timeouts

marcuseide committed:
Filter out undefined keys from settings

Signed-off-by: Marcus Eide <eide@spotify.com>

marcuseide committed:
Add changeset

Signed-off-by: Marcus Eide <eide@spotify.com>

Commit:
Added runson command to fix the workflow config

Signed-off-by: Harry Hogg <hhogg@spotify.com>

freben committed:
Merge pull request #7915 from backstage/freben/kubernetes-config-test

get rid of config loading here too

freben committed:
Merge pull request #7918 from backstage/fix-snyk-github-issue-workflow

Added runson command to fix the snyk github issue sync workflow config

freben committed:
Merge pull request #7906 from backstage/dependabot/npm_and_yarn/apollo-server-2.25.3

build(deps): bump apollo-server from 2.21.0 to 2.25.3

freben committed:
Merge pull request #7910 from jeremyguarini/fix_code_coverage_chart_xaxis

display X-axis in ascending order

Commit:
Fixes the Snyk github issue workflow to continue when the Snyk CLI exits with an error

Signed-off-by: Harry Hogg <hhogg@spotify.com>

Rugvip committed:
Merge pull request #7830 from kennethzfeng/kfeng.is-execute

plugins/scaffolder-backend: Fix a bug where only file mode 775 is con…

egnwd committed:
Allow for material UI cellStyle function from the backstage Table component

Signed-off-by: Elliot Greenwood <elliot@fundapps.co>

egnwd committed:
Add StyledTable to the Table stories

Signed-off-by: Elliot Greenwood <elliot@fundapps.co>

jhaals committed:
Merge pull request #7913 from backstage/jhaals/core-plugin-deps

core-plugin-api: Deprecate unused types

jhaals committed:
Merge pull request #7919 from backstage/fix-snyk-github-issue-workflow-2

Fixes the Snyk github issue workflow to continue when the Snyk CLI ex…

benjdlambert committed:
Merge pull request #7883 from GoFightNguyen/patch-1

[docs] clarify example showing that parameters types propagate

egnwd committed:
Add changeset

Signed-off-by: Elliot Greenwood <elliot@fundapps.co>

benjdlambert committed:
Merge pull request #7825 from alexcurtin/catalog-client-api-ref

Adding API documentation for catalog-client

freben committed:
revert the graphql package pinning

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

Rugvip committed:
Merge pull request #7907 from backstage/rugvip/nonoprivate

create-app: default to making new plugins private

JoThomsen committed:
Update ADOPTERS.md

benjdlambert committed:
Merge pull request #7672 from marleypowell/marley/7641-consume-exported-ado-types

refactor(`@backstage/plugin-azure-devops`): Consume re-exported types from `@backstage/plugin-azure-devops-common`.

benjdlambert committed:
Merge pull request #7920 from backstage/freben/drop-overrides

revert the graphql package pinning

freben committed:
Merge pull request #7923 from JoThomsen/patch-1

Added Signal Iduna to ADOPTERS.md

Rugvip committed:
auth-backend: add forwarding of the SAML audience option

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

freben committed:
Merge pull request #7916 from backstage/eide/firestore-clean-settings

[auth-backend] Only use settings that have a value when creating a FirestoreKeyStore

Rugvip committed:
Merge pull request #7924 from backstage/rugvip/audience

auth-backend: add forwarding of the SAML audience option

Rugvip committed:
add changeset for @spotify/eslint-config-typescript bump

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
Merge pull request #7890 from timja/adjust-graph-processor

Tweak logic for msgraph catalog ingesting for display names with security groups

Rugvip committed:
Merge pull request #7912 from RodionGork/fix/kuber-plugin-pass-id-token-on-getclusters

fix: kuber plugin pass id token on getClusters req

Rugvip committed:
add changeset for @spotify/eslint-config-react bump

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
Merge pull request #7833 from backstage/dependabot/npm_and_yarn/spotify/eslint-config-typescript-12.0.0

build(deps): bump @spotify/eslint-config-typescript from 10.0.0 to 12.0.0

Rugvip committed:
Merge pull request #7921 from egnwd/core-table-cellStyle-fn

[core-components] Allow for material-table cellStyle function from the backstage Table component

cmpadden committed:
note internal refactor of LocalPublish changes and impact on usage

Signed-off-by: Colton Padden <colton.padden@fastmail.com>

Rugvip committed:
Merge pull request #7851 from cmpadden/publisher-object-pattern

Use Parameter Objects for Publisher Constructors

Commit:
swap out data-grid for material-table

Signed-off-by: Jeremy Guarini <jguarini@paloaltonetworks.com>

Rugvip committed:
core-app-api: add missing core feature icons + move icon things around a bit

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

jhaals committed:
core-plugin-api: Deprecate use of extensions without name

Signed-off-by: Johan Haals <johan.haals@gmail.com>

Commit:
Added ts-node as a dependency for the Snyk Github issues sync script

Signed-off-by: Harry Hogg <hhogg@spotify.com>

jhaals committed:
api report

Signed-off-by: Johan Haals <johan.haals@gmail.com>

Rugvip committed:
app-defaults: bump internal deps

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

freben committed:
Merge pull request #7933 from backstage/fix-snyk-github-issue-workflow-3

Added ts-node as a dependency for the Snyk Github issues sync script

Rugvip committed:
core-app-api: createSpecializedApp returns a BackstageApp

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
core-app-api: switch auth provider icons to be empty by default

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

jhaals committed:
Update docs/features/software-catalog/catalog-customization.md

Signed-off-by: Johan Haals <johan.haals@gmail.com>

Co-authored-by: Fredrik Adelöw <freben@gmail.com>

jhaals committed:
Update .changeset/quiet-clocks-push.md

Signed-off-by: Johan Haals <johan.haals@gmail.com>

Co-authored-by: Fredrik Adelöw <freben@gmail.com>

Rugvip committed:
create-app: add app-defaults dependency to app + notice in changeset

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
docs: introduce deprecation docs + app-theme deprecation

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
core-app-api,app-defaults: API report and doc fixups

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
app-defaults: clean up dependencies

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
changesets: add changesets for test-utils createApp and AppTheme deprecation

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Commit:
Install dependeices before trying to use them in the Snyk github issue workflow

Signed-off-by: Harry Hogg <hhogg@spotify.com>

freben committed:
Remove the unused UserFlags type

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

Rugvip committed:
cli: fix for config:check only validating frontend config

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
test-utils: switch mock icons to be plain svg elements

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

freben committed:
Merge pull request #7934 from backstage/fix-snyk-github-issue-workflow-4

Install dependeices before trying to use them in the Snyk github issu…

Rugvip committed:
core-app-api: fix fix docs docs

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
test-utils: reintroduce defaultApis

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Commit:
Rename gauge props to GaugeProps and export

Signed-off-by: James Turley <jamesturley@gocardless.com>

Commit:
Only return string from color getter type

Signed-off-by: James Turley <jamesturley@gocardless.com>

Commit:
Fixup and rename Gauge color getter function types

Signed-off-by: James Turley <jamesturley@gocardless.com>

Commit:
Update changeset to reflect naming changes

Signed-off-by: James Turley <jamesturley@gocardless.com>

timbonicus committed:
Add missing link to Gitlab discovery

Signed-off-by: Tim Hansen <timbonicus@gmail.com>

timbonicus committed:
Merge pull request #7938 from backstage/timbonicus/missing-gitlab-link

Add missing link to Gitlab discovery

Commit:
Update API report

Signed-off-by: James Turley <jamesturley@gocardless.com>

Rugvip committed:
app-defaults: lazier api initialization to work around import cycle

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
core-app-api: fix AppManager tests

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
Merge pull request #7937 from backstage/rugvip/configcheck

cli: fix for config:check only validating frontend config

Rugvip committed:
test-utils: migrate theme usage and switch to actual MUI icons

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
Merge pull request #7732 from tragiclifestories/gauge-color

Allow overriding colour in Gauge components

Commit:
Use string for Date and let client handle display

Signed-off-by: Andre Wanlin <awanlin@rapidrtc.com>

awanlin committed:
Merge branch 'backstage:master' into topic/improve-azure-devops-date-handling

dependabot[bot] committed:
build(deps): bump @rollup/plugin-commonjs from 17.1.0 to 21.0.1

Bumps [@rollup/plugin-commonjs](https://github.com/rollup/plugins/tree/HEAD/packages/commonjs) from 17.1.0 to 21.0.1.
- [Release notes](https://github.com/rollup/plugins/releases)
- [Changelog](https://github.com/rollup/plugins/blob/master/packages/commonjs/CHANGELOG.md)
- [Commits](https://github.com/rollup/plugins/commits/commonjs-v21.0.1/packages/commonjs)

---
updated-dependencies:
- dependency-name: "@rollup/plugin-commonjs"
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

HHogg committed:
Pass the GH token as an ENV to the Github step of the Snyk workflow

Signed-off-by: Harry Hogg <harry@hogg.io>

rodmachen committed:
Add Search Modal component (#7316)

Signed-off-by: rodmachen <rmachen@expediagroup.com>

freben committed:
Merge pull request #7932 from backstage/jhaals/core-plugin-api-no-name

core-plugin-api: Deprecate use of extensions without name

Rugvip committed:
Merge pull request #7945 from backstage/hhogg/fix-snyk-github-issue-workflow-5

Pass the GH token as an ENV to the Github step of the Snyk workflow

Rugvip committed:
Merge pull request #7911 from backstage/dependabot/npm_and_yarn/spotify/eslint-config-react-12.0.0

build(deps): bump @spotify/eslint-config-react from 10.0.0 to 12.0.0

Fox32 committed:
Fix styling issues of swagger-ui

Signed-off-by: Oliver Sand <oliver.sand@sda-se.com>

kim5566 committed:
Update TechDocs Headings css font weight (#7931)

* update headings style

Signed-off-by: kim5566 <28945404+kim5566@users.noreply.github.com>

* add changeset

Signed-off-by: kim5566 <28945404+kim5566@users.noreply.github.com>

* update changeset

Signed-off-by: kim5566 <28945404+kim5566@users.noreply.github.com>

* rename changeset

Signed-off-by: kim5566 <28945404+kim5566@users.noreply.github.com>

freben committed:
Make sure that the server builder propagates init errors properly

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

Fox32 committed:
Merge pull request #7865 from SDA-SE/feat/api-docs-customization

Lazy load API definition widgets

github-actions[bot] committed:
Version Packages

freben committed:
bump uglify-js

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
bump refractor and, hence, prismjs

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
bump xss

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

Rugvip committed:
Merge pull request #7823 from backstage/changeset-release/master

Version Packages

freben committed:
bump @npmcli/git

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

Commit:
Fixes a minor markdown foobar in the body output

Signed-off-by: Harry Hogg <hhogg@spotify.com>

freben committed:
Merge pull request #7981 from backstage/freben/bump-uglify-js

bump uglify-js

freben committed:
Merge pull request #7982 from backstage/freben/bump-prismjs

bump refractor and, hence, prismjs

freben committed:
partially bump nth-check

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
bump prompts

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
Merge pull request #7983 from backstage/freben/bump-xss

bump xss

freben committed:
Merge pull request #7985 from backstage/freben/bump-npmcli-git

bump @npmcli/git

freben committed:
Merge pull request #7987 from backstage/freben/bump-nth-check

partially bump nth-check

freben committed:
Merge pull request #7988 from backstage/freben/bump-prompts

bump prompts

freben committed:
add snyk ignores for the node-gyp tar vuln

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
Merge pull request #7935 from backstage/freben/userflags

Remove the unused UserFlags type

awanlin committed:
Merge branch 'backstage:master' into topic/improve-azure-devops-date-handling

Rugvip committed:
Merge pull request #7986 from backstage/hhogg/fix-snyk-github-issue-script

Fixes a minor markdown foobar in the body output

Commit:
Cleaned up tests

Signed-off-by: Andre Wanlin <awanlin@rapidrtc.com>

Commit:
Improved readability and used DateTime consistently

Signed-off-by: Andre Wanlin <awanlin@rapidrtc.com>

Commit:
Updated story to handle date changes

Signed-off-by: Andre Wanlin <awanlin@rapidrtc.com>

Commit:
Aded API Report and changeset

Signed-off-by: Andre Wanlin <awanlin@rapidrtc.com>

marleypowell committed:
fix: Fixed issue with running a subset of stories caused by incorrectly resolved paths.

Signed-off-by: Marley Powell <Marley.Powell@exclaimer.com>

Rugvip committed:
app-defaults: update internal deps

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
cli: make sure JSS styles receive the highest priority

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
Merge pull request #7790 from backstage/mob/decouple-core

core-app-api: deprecate dependency on core-components, theme, and MUI

freben committed:
review comments

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
fixup yarn.lock

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

freben committed:
Merge pull request #7990 from backstage/freben/tar2

add snyk ignores for the node-gyp tar vuln

Commit:
swapout material-table for backstage table

Signed-off-by: Jeremy Guarini <jguarini@paloaltonetworks.com>

Commit:
ran yarn install to clear out data-grid dependency

Signed-off-by: Jeremy Guarini <jguarini@paloaltonetworks.com>

Rugvip committed:
Merge pull request #7845 from jeremyguarini/gcp-plugin-ui-updates

UI updates to GCP-projects plugin

Rugvip committed:
Merge pull request #8001 from backstage/rugvip/oda

cli: make sure JSS styles receive the highest priority

dependabot[bot] committed:
build(deps-dev): bump @types/jquery from 3.5.5 to 3.5.8

Bumps [@types/jquery](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/jquery) from 3.5.5 to 3.5.8.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/jquery)

---
updated-dependencies:
- dependency-name: "@types/jquery"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/nunjucks from 3.1.5 to 3.2.0

Bumps [@types/nunjucks](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/nunjucks) from 3.1.5 to 3.2.0.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/nunjucks)

---
updated-dependencies:
- dependency-name: "@types/nunjucks"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

marleypowell committed:
Merge branch 'master' of https://github.com/backstage/backstage into marley/7999-storybook-fix

marleypowell committed:
refactor: Reworked how subset story paths are created.

Signed-off-by: Marley Powell <Marley.Powell@exclaimer.com>

freben committed:
Merge pull request #7947 from backstage/freben/service-builder-errors

Make sure that the server builder propagates init errors properly

jhaals committed:
Merge pull request #8015 from backstage/dependabot/npm_and_yarn/types/jquery-3.5.8

build(deps-dev): bump @types/jquery from 3.5.5 to 3.5.8

jhaals committed:
Merge pull request #8016 from backstage/dependabot/npm_and_yarn/types/nunjucks-3.2.0

build(deps-dev): bump @types/nunjucks from 3.1.5 to 3.2.0

Rugvip committed:
Merge pull request #8000 from marleypowell/marley/7999-storybook-fix

fix: Fixed issue with running a subset of stories caused by incorrectly resolved paths.

Fox32 committed:
Add theme switcher to sidebar of dev app

Signed-off-by: Oliver Sand <oliver.sand@sda-se.com>

iain-b committed:
Dynamic annotations

Signed-off-by: Iain Billett <iain@roadie.io>

iain-b commented:
> The CI thing is interesting. I agree we shouldn't hard code those but I'm not certain what is a better approach. Maybe it just needs to be a fact retriever for that specific plugin? I feel like that would be the correct way to do it tbh.

Yeah I think you're right. In the meantime I've allowed us to configure a list of annotations we want facts for when we instantiate the retriever. My thinking is this will allow us do do checks around annotations for now until plugin devs create their own retrievers.

mtlewis committed:
build(deps): bump @microsoft/api-documenter to v7.13.68

Signed-off-by: Mike Lewis <mtlewis@users.noreply.github.com>

benjdlambert committed:
Merge pull request #8021 from SDA-SE/feat/dev-switch-theme

Add theme switcher to sidebar of dev app

freben committed:
Merge pull request #7612 from backstage/freben/recurring-tasks

Add support for distributed tasks

awanlin committed:
Merge branch 'backstage:master' into topic/improve-azure-devops-date-handling

benjdlambert committed:
Add changeset

Signed-off-by: Ben Lambert <blam@spotify.com>

mtlewis committed:
Merge pull request #8023 from backstage/bump-api-documenter

build(deps): bump @microsoft/api-documenter to v7.13.68

Commit:
Updated changeset to minor for common

Signed-off-by: Andre Wanlin <awanlin@rapidrtc.com>

benjdlambert committed:
Merge pull request #7998 from awanlin/topic/improve-azure-devops-date-handling

Improved Azure DevOps Date Handling

Commit:
Skip empty file names during scaffolder

Signed-off-by: Tim Jacomb <tim.jacomb@hmcts.net>

zfalen-deloitte committed:
hotfix for Backstage token generation, prefer .token over .idToken

Signed-off-by: Zach Falen <zfalen@deloitte.com>

zfalen-deloitte committed:
add changeset

Signed-off-by: Zach Falen <zfalen@deloitte.com>

zfalen-deloitte committed:
update test

Signed-off-by: Zach Falen <zfalen@deloitte.com>

Rugvip committed:
scripts: make snyk issue script executable

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
scripts: tweak snyk script to match license issues

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

freben committed:
Merge pull request #8037 from backstage/rugvip/issuefix

scripts: tweak snyk issue script to catch existing license issues

Rugvip committed:
cli: fix for backend bundle building all packages

Signed-off-by: Patrik Oldsberg <poldsberg@gmail.com>

Rugvip committed:
Merge pull request #8041 from backstage/rugvip/ALL-THE-PACKAGES

cli: fix for backend bundle building all packages

dependabot[bot] committed:
build(deps-dev): bump eslint-plugin-cypress from 2.11.3 to 2.12.1

Bumps [eslint-plugin-cypress](https://github.com/cypress-io/eslint-plugin-cypress) from 2.11.3 to 2.12.1.
- [Release notes](https://github.com/cypress-io/eslint-plugin-cypress/releases)
- [Commits](https://github.com/cypress-io/eslint-plugin-cypress/compare/v2.11.3...v2.12.1)

---
updated-dependencies:
- dependency-name: eslint-plugin-cypress
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump js-base64 from 3.6.1 to 3.7.2

Bumps [js-base64](https://github.com/dankogai/js-base64) from 3.6.1 to 3.7.2.
- [Release notes](https://github.com/dankogai/js-base64/releases)
- [Commits](https://github.com/dankogai/js-base64/compare/3.6.1...3.7.2)

---
updated-dependencies:
- dependency-name: js-base64
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

tbenade committed:
small grammatical error

Signed-off-by: Terrence Benade <tbenade@seek.com.au>

dehamzah committed:
update api report docs

Signed-off-by: Dede Hamzah <dehamzah@gmail.com>

freben committed:
Merge pull request #8046 from tbenade/faq-correction

small grammatical error

freben committed:
Merge pull request #8045 from backstage/dependabot/npm_and_yarn/js-base64-3.7.2

build(deps): bump js-base64 from 3.6.1 to 3.7.2

jhaals committed:
Merge pull request #8044 from backstage/dependabot/npm_and_yarn/eslint-plugin-cypress-2.12.1

build(deps-dev): bump eslint-plugin-cypress from 2.11.3 to 2.12.1

punkle committed:
update versions of json-schema

Signed-off-by: Brian Fletcher <brian@roadie.io>

jhaals committed:
Merge pull request #8048 from RoadieHQ/update-json-schema

update versions of json-schema

jhaals committed:
Merge pull request #7885 from bryce-od/defaultProcessors

feat: Add getDefaultProcessors to CatalogBuilder

Rugvip committed:
Merge pull request #8035 from zfalen-deloitte/bugfix/duplicate-backstage-token

Bugfix/duplicate backstage token

Rugvip committed:
Merge pull request #7944 from backstage/dependabot/npm_and_yarn/rollup/plugin-commonjs-21.0.1

build(deps): bump @rollup/plugin-commonjs from 17.1.0 to 21.0.1

Rugvip committed:
Merge pull request #7844 from cmpadden/create-app-task-test

Create App Refactoring and Unit Testing

Rugvip committed:
Merge pull request #7886 from SDA-SE/feat/about-api-component-only

feat(catalog): remove "View Api" from AboutCard

Rugvip committed:
Merge pull request #7874 from dehamzah/override-icons

Add Props Icon for Sidebar Item SidebarSearchField and Settings

iain-b committed:
Add hasTags schema

benjdlambert committed:
Merge pull request #8028 from timja/skip-empty-file-name-scaffolder

Skip empty file names during scaffolder

jhaals committed:
Add missing yarn.lock changes and reports

Signed-off-by: Johan Haals <johan.haals@gmail.com>

jhaals committed:
Merge pull request #8052 from backstage/jhaals/bump-yarn

Add missing yarn.lock changes and reports

iain-b committed:
Expose catalogFactRetriever & configure in backend

Signed-off-by: Iain Billett <iain@roadie.io>
